src/modules/navigator/mission.cpp

```diff
@@ -733,14 +737,45 @@ Mission::set_mission_items()
 			_mission_item.autocontinue = true;
 			_mission_item.time_inside = 0.0f;
 			_mission_item.disable_mc_yaw = true;
+		} else if (_mission_item.nav_cmd == NAV_CMD_LAND && _work_item_type == WORK_ITEM_TYPE_DEFAULT) {
+			landing = true;
 		}
 
 		/* we just moved to the landing waypoint, now descend */
 		if (_work_item_type == WORK_ITEM_TYPE_MOVE_TO_LAND &&
 		    new_work_item_type == WORK_ITEM_TYPE_DEFAULT) {
 
 			new_work_item_type = WORK_ITEM_TYPE_DEFAULT;
-			/* XXX: noop */
+
+			landing = true;
+
+		}
+
+		// switch to precision land if set in the mission item
+		if (landing && _mission_item.land_precision > 0)
+		{
+			struct vehicle_command_s cmd = {};
+			cmd.timestamp = hrt_absolute_time();;
```

Comment:
this jump from the mission to the commander could be avoided somehow? I think we could do something to split this and not having things messed up

Reply:
Would be nice if it could be avoided. Any ideas how?

Reply:
probably a direct call to `PrecLand`?

Reply:
We could have the same implementation but specific for land during mission and land command.

Reply:
I switched to direct calls to `PrecLand` in https://github.com/PX4/Firmware/pull/8160/commits/3ebd33a72be316686efdc0ec1dc4257fb0a56ab7, what do you think?

Reply:
This looks much more cleaner now. Great work @ndepal!